Assets/SEE/Utils/GraphUtils.cs

```diff
@@ -0,0 +1,105 @@
+                SourceName = name,
+                ID = id,
+                Type = type
+            };
+
+            graph.AddNode(result);
+            return result;
+        }
+
+        /// <summary>
+        /// Creates a new node for each element of a filepath, that does not
+        /// already exists in the graph.
+        /// </summary>
+        /// <param name="path">The remaining part of the path</param>
+        /// <param name="parent">The parent node from the current element of the path</param>
+        /// <param name="parentPath">The path of the current parent, which will eventually be part of the ID</param>
+        /// <param name="graph">The graph to which the new node belongs to</param>
+        /// <param name="mainNode">The root node of the main directory</param>
+        public static Node BuildGraphFromPath(string path, Node parent, string parentPath, Graph graph, Node mainNode)
+        {
+            string[] pathSegments = path.Split(Path.AltDirectorySeparatorChar);
+            string nodePath = string.Join(Path.AltDirectorySeparatorChar.ToString(), pathSegments, 1,
+                pathSegments.Length - 1);
+            // Current pathSegment is in the main directory.
+            if (parentPath == null)
+            {
+                // Directory already exists.
+                if (graph.GetNode(pathSegments[0]) != null)
+                {
+                    return BuildGraphFromPath(nodePath, graph.GetNode(pathSegments[0]), pathSegments[0], graph,
+                        mainNode);
+                }
+
+                // Directory does not exist.
+                if (graph.GetNode(pathSegments[0]) == null && pathSegments.Length > 1 && parent == null)
+                {
+                    mainNode.AddChild(NewNode(graph, pathSegments[0], "directory", pathSegments[0]));
+                    return BuildGraphFromPath(nodePath, graph.GetNode(pathSegments[0]), pathSegments[0], graph,
+                        mainNode);
+                }
+
+                // I dont know, if this code ever gets used -> I dont know, how to handle empty directorys.
+                if (graph.GetNode(pathSegments[0]) == null && pathSegments.Length == 1 && parent == null)
+                {
+                    mainNode.AddChild(NewNode(graph, pathSegments[0], "directory", pathSegments[0]));
+                    return null;
+                }
+            }
+
+            // Current pathSegment is not in the main directory.
+            if (parentPath != null)
+            {
+                // The node for the current pathSegment exists.
+                if (graph.GetNode(parentPath + Path.DirectorySeparatorChar + pathSegments[0]) != null)
+                {
+                    return BuildGraphFromPath(nodePath,
+                        graph.GetNode(parentPath + Path.DirectorySeparatorChar + pathSegments[0]),
+                        parentPath + Path.DirectorySeparatorChar + pathSegments[0], graph, mainNode);
```

Comment:
You could separate out `parentPath + Path.DirectorySeparatorChar + pathSegments[0]` into a string variable first. Also applies to code blocks below.